Bottom: e4830778fdf80c34ddf221a8898db1d6615f24d4
Top:    2fa435f9a212c1ef1e67f17f89976d5680fbd472
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-13 01:40:18 +0100

Don't make GUI drawing calls to erase objects

Changing the colour here is unnecessary (as drawing is deferred).
Delete the gui->set_color() calls, and additionally, comment out
all real-time drawing frmo the autorouter. This hasn't been ported
to be compatible with the GL code, which requires specific setup
and teardown around drawing.


---

diff --git a/src/autoroute.c b/src/autoroute.c
index 7fb7443..3361f79 100644
--- a/src/autoroute.c
+++ b/src/autoroute.c
@@ -1421,9 +1421,11 @@ bloat_routebox (routebox_t * rb)
 void
 fillbox (const BoxType * b)
 {
+#if 0
   LayerTypePtr SLayer = LAYER_PTR (0);
   gui->set_color (Output.fgGC, SLayer->Color);
   gui->fill_rect (Output.fgGC, b->X1, b->Y1, b->X2, b->Y2);
+#endif
 }
 
 /* makes a line on the solder layer silk surrounding the box */
@@ -1437,7 +1439,7 @@ showbox (BoxType b, Dimension thickness, int group)
   if (showboxen != -1 && showboxen != group)
     return;
 
-
+#if 0
   gui->set_line_width (Output.fgGC, thickness);
   gui->set_line_cap (Output.fgGC, Trace_Cap);
   gui->set_color (Output.fgGC, SLayer->Color);
@@ -1447,6 +1449,7 @@ showbox (BoxType b, Dimension thickness, int group)
   gui->draw_line (Output.fgGC, b.X1, b.Y1, b.X1, b.Y2);
   gui->draw_line (Output.fgGC, b.X2, b.Y1, b.X2, b.Y2);
   gui->use_mask (HID_FLUSH_DRAW_Q);
+#endif
 
 #if 1
   if (b.Y1 == b.Y2 || b.X1 == b.X2)
@@ -1491,6 +1494,7 @@ showedge (edge_t * e)
 {
   BoxType *b = (BoxType *) e->rb;
 
+#if 0
   gui->set_line_cap (Output.fgGC, Trace_Cap);
   gui->set_line_width (Output.fgGC, 1);
   gui->set_color (Output.fgGC, Settings.MaskColor);
@@ -1512,6 +1516,7 @@ showedge (edge_t * e)
     default:
       break;
     }
+#endif
 }
 #endif
 
@@ -1545,9 +1550,11 @@ EraseRouteBox (routebox_t * rb)
       X2 = rb->box.X2 - thick / 2;
     }
 
+#if 0
   gui->set_line_width (ar_gc, thick);
   gui->set_color (ar_gc, Settings.BackgroundColor);
   gui->draw_line (ar_gc, X1, Y1, X2, Y2);
+#endif
 }
 
 /* return a "parent" of this edge which immediately precedes it in the route.*/
@@ -3235,11 +3242,13 @@ RD_DrawVia (routedata_t * rd, LocationType X, LocationType Y,
       r_insert_entry (rd->layergrouptree[rb->group], &rb->box, 1);
       rb->flags.homeless = 0;	/* not homeless anymore */
 
+#if 0
       if (TEST_FLAG (LIVEROUTEFLAG, PCB))
 	{
 	  gui->set_color (ar_gc, PCB->ViaColor);
 	  gui->fill_circle (ar_gc, X, Y, radius);
 	}
+#endif
     }
 }
 static void
@@ -3327,6 +3336,7 @@ RD_DrawLine (routedata_t * rd,
   /* and add it to the r-tree! */
   r_insert_entry (rd->layergrouptree[rb->group], &rb->box, 1);
 
+#if 0
   if (TEST_FLAG (LIVEROUTEFLAG, PCB))
     {
       LayerTypePtr layp = LAYER_PTR (PCB->LayerGroups.Entries[rb->group][0]);
@@ -3335,6 +3345,7 @@ RD_DrawLine (routedata_t * rd,
       gui->set_color (ar_gc, layp->Color);
       gui->draw_line (ar_gc, qX1, qY1, qX2, qY2);
     }
+#endif
 
   /* and to the via space structures */
   if (AutoRouteParameters.use_vias)
@@ -3650,8 +3661,10 @@ TracePath (routedata_t * rd, routebox_t * path, const routebox_t * target,
   while (!path->flags.source);
   /* flush the line queue */
   RD_DrawLine (rd, -1, 0, 0, 0, 0, 0, NULL, false, false);
+#if 0
   if (TEST_FLAG (LIVEROUTEFLAG, PCB))
     gui->use_mask (HID_FLUSH_DRAW_Q);
+#endif
 }
 
 /* create a fake "edge" used to defer via site searching. */
@@ -3950,8 +3963,8 @@ __show_tree (const BoxType * b, void *cl)
 static void
 show_tree (rtree_t * tree, int even_odd)
 {
-  r_search (tree, NULL, NULL, __show_tree, (void *) even_odd);
-  gui->use_mask (HID_FLUSH_DRAW_Q);
+//  r_search (tree, NULL, NULL, __show_tree, (void *) even_odd);
+//  gui->use_mask (HID_FLUSH_DRAW_Q);
 }
 
 #endif
@@ -4730,8 +4743,8 @@ RouteAll (routedata_t * rd)
 		    }
 		}
 	      END_LOOP;
-	      if (TEST_FLAG (LIVEROUTEFLAG, PCB))
-		gui->use_mask (HID_FLUSH_DRAW_Q);
+//	      if (TEST_FLAG (LIVEROUTEFLAG, PCB))
+//		gui->use_mask (HID_FLUSH_DRAW_Q);
 	      /* reset to original connectivity */
 	      if (rip)
 		{
@@ -5089,16 +5102,20 @@ AutoRoute (bool selected)
 
   total_wire_length = 0;
   total_via_count = 0;
+#if 0
   if (TEST_FLAG (LIVEROUTEFLAG, PCB))
     {
       gui->use_mask (HID_LIVE_DRAWING);
     }
+#endif
 
+#if 0
   if (ar_gc == 0)
     {
       ar_gc = gui->make_gc ();
       gui->set_line_cap (ar_gc, Round_Cap);
     }
+#endif
 
   for (i = 0; i < NUM_STYLES; i++)
     {
@@ -5258,10 +5275,12 @@ donerouting:
   Message ("Total added wire length = %f inches, %d vias added\n",
 	   total_wire_length / 1.e5, total_via_count);
   DestroyRouteData (&rd);
+#if 0
   if (TEST_FLAG (LIVEROUTEFLAG, PCB))
     {
       gui->use_mask (HID_LIVE_DRAWING_OFF);
     }
+#endif
   if (changed)
     {
       SaveUndoSerialNumber ();
diff --git a/src/draw.c b/src/draw.c
index baca2b5..9916307 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2152,7 +2152,6 @@ void
 EraseVia (PinTypePtr Via)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPinOrViaLowLevel (Via, false);
   if (TEST_FLAG (DISPLAYNAMEFLAG, Via))
     DrawPinOrViaNameLowLevel (Via);
@@ -2166,7 +2165,6 @@ void
 EraseRat (RatTypePtr Rat)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   if (TEST_FLAG(VIAFLAG, Rat))
     {
       int w = Rat->Thickness;
@@ -2191,7 +2189,6 @@ void
 EraseViaName (PinTypePtr Via)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPinOrViaNameLowLevel (Via);
   Erasing--;
 }
@@ -2203,7 +2200,6 @@ void
 ErasePad (PadTypePtr Pad)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPadLowLevel (Output.fgGC, Pad, false, false);
   if (TEST_FLAG (DISPLAYNAMEFLAG, Pad))
     DrawPadNameLowLevel (Pad);
@@ -2217,7 +2213,6 @@ void
 ErasePadName (PadTypePtr Pad)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPadNameLowLevel (Pad);
   Erasing--;
 }
@@ -2229,7 +2224,6 @@ void
 ErasePin (PinTypePtr Pin)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPinOrViaLowLevel (Pin, false);
   if (TEST_FLAG (DISPLAYNAMEFLAG, Pin))
     DrawPinOrViaNameLowLevel (Pin);
@@ -2243,7 +2237,6 @@ void
 ErasePinName (PinTypePtr Pin)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPinOrViaNameLowLevel (Pin);
   Erasing--;
 }
@@ -2255,7 +2248,6 @@ void
 EraseLine (LineTypePtr Line)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawLineLowLevel (Line, false);
   Erasing--;
 }
@@ -2269,7 +2261,6 @@ EraseArc (ArcTypePtr Arc)
   if (!Arc->Thickness)
     return;
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawArcLowLevel (Arc);
   Erasing--;
 }
@@ -2282,7 +2273,6 @@ EraseText (LayerTypePtr Layer, TextTypePtr Text)
 {
   int min_silk_line;
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   if (Layer == & PCB->Data->SILKLAYER
       || Layer == & PCB->Data->BACKSILKLAYER)
     min_silk_line = PCB->minSlk;
@@ -2299,7 +2289,6 @@ void
 ErasePolygon (PolygonTypePtr Polygon)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawPolygonLowLevel (Polygon);
   Erasing--;
 }
@@ -2311,8 +2300,6 @@ void
 EraseElement (ElementTypePtr Element)
 {
   Erasing++;
-  /* set color and draw lines, arcs, text and pins */
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   ELEMENTLINE_LOOP (Element);
   {
     DrawLineLowLevel (line, false);
@@ -2336,7 +2323,6 @@ void
 EraseElementPinsAndPads (ElementTypePtr Element)
 {
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   PIN_LOOP (Element);
   {
     DrawPinOrViaLowLevel (pin, false);
@@ -2363,7 +2349,6 @@ EraseElementName (ElementTypePtr Element)
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
   Erasing++;
-  gui->set_color (Output.fgGC, Settings.BackgroundColor);
   DrawStrippedText (Element, PCB->minSlk);
   Erasing--;
 }
